### PR TITLE
feat: support Service Catalog AppRegistry

### PR DESCRIPTION
**Pull request by jingnanl**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

Update pipeline workflow and stacks to support the feature of associating Service Catalog AppRegistry application with stacks created by the pipeline.

Upgrade jacoco to 1.6.1 for `TypeError: Cannot read properties of undefined (reading 'split')` (https://github.com/Madrapps/jacoco-report/releases/tag/v1.6.1)

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [x] add new test cases
- [ ] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

File: src/control-plane/backend/lambda/api/common/constants.ts [hunk 1]
```diff
@@ -62,9 +62,30 @@ const PIPELINE_SUPPORTED_REGIONS = [
   'cn-northwest-1',
 ];
 
+const SERVICE_CATALOG_SUPPORTED_REGIONS = [
```
Comment (zxkane) [inline src/control-plane/backend/lambda/api/common/constants.ts:65]: The backend is sharing the infra's constants via file `constants-ln`, could you move those to the infra's constants to share this definition?

File: src/control-plane/backend/lambda/api/common/utils.ts [hunk 2]
```diff
@@ -835,6 +843,11 @@ function pathNodesToAttribute(nodes: IMetadataRawValue[] | undefined) {
   return pathNodes;
 }
 
+function getAppRegistryApplicationArn(pipeline: IPipeline): string {
+  return SERVICE_CATALOG_SUPPORTED_REGIONS.includes(pipeline.region) ?
```
Comment (zxkane) [inline src/control-plane/backend/lambda/api/common/utils.ts:847]: How about we always fetch the ARN of the appregsitry application? If it's not found, the value is empty.

File: src/control-plane/backend/stack-action-state-machine-construct.ts [hunk 3]
```diff
@@ -165,6 +165,7 @@ export class StackActionStateMachine extends Construct {
             'globalaccelerator:*',
             'kms:*',
             'athena:*',
+            'servicecatalog:*',
```
Comment (zxkane) [inline src/control-plane/backend/stack-action-state-machine-construct.ts:168]: Let's grant the least privileges to the service catalog.

**Review by zxkane — CHANGES_REQUESTED** [review]

File: src/control-plane/backend/lambda/api/common/constants.ts (hunk 1 above)
Comment (jingnanl) [reply src/control-plane/backend/lambda/api/common/constants.ts]: done

File: src/control-plane/backend/stack-action-state-machine-construct.ts (hunk 3 above)
Comment (jingnanl) [reply src/control-plane/backend/stack-action-state-machine-construct.ts]: done

File: src/control-plane/backend/lambda/api/common/utils.ts (hunk 2 above)
Comment (jingnanl) [reply src/control-plane/backend/lambda/api/common/utils.ts:848]: update the workflow to be consistent in all the regions

File: src/common/parameters.ts [hunk 4]
```diff
@@ -454,5 +455,13 @@ export class Parameters {
       appIdsParam,
     };
   }
-}
 
+  public static createAppRegistryApplicationArnParameters(scope: Construct, id?: string) : CfnParameter {
+    return new CfnParameter(scope, id ?? 'AppRegistryApplicationArn', {
+      description: 'Service Catalog AppRegistry Application Arn',
+      type: 'String',
+      allowedPattern: `^(|${SERVICE_CATALOG_APPREGISTRY_ARN_PATTERN})$`,
```
[the remark this answers is not in the document]
Comment (jingnanl) [reply src/common/parameters.ts:464]: done

**Review by zxkane — APPROVED** [review]

**Review by tyyzqmf — APPROVED** [review]

**Pull request merged**